VirtoCommerce.Storefront/Controllers/Api/ApiOrderController.cs

```diff
@@ -180,6 +180,33 @@ public async Task<ActionResult> GetInvoicePdf(string orderNumber)
             return File(stream, "application/pdf");
         }
 
+        // PUT: storefrontapi/orders/{orderNumber}/status
+        [HttpPut("{orderNumber}/status")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> ChangeOrderStatus(string orderNumber, [FromBody] OrderStatusChange payload)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            using (await AsyncLock.GetLockByKey(GetAsyncLockKey(orderNumber, WorkContext)).LockAsync())
+            {
+                var order = await _orderApi.GetByNumberAsync(orderNumber);
```

Comment:
You are loading order twice. Is it really necessary?  I can't see any obstacles to check authorization  for order loaded by call `GetOrderDtoByNumber` method